Changes required for running on the CTV signet.

fiatjaf opened the pull request:
- remove useless SelectParams call since bitcoinlib doesn't know signet.
- remove generateblocks, replace it with a message telling the user to get signet coins.
- hardcode demo scenario to signet.
- output the "txid:output" as the original_coin instead of just the txid.
- update readme.

fiatjaf committed:
gitignore pycache.

jamesob commented on README.md:
```diff
@@ -1,7 +1,7 @@
 
 This repository demonstrates an implementation of simple, "single-hop" vaults
-using the proposed `OP_CHECKTEMPLATEVERIFY` opcode. 
+using the proposed `OP_CHECKTEMPLATEVERIFY` opcode.
```
Would've preferred these doc/whitespace changes in a separate commit, but no big deal.

fiatjaf replied:
Sorry, my editor did this and I know it sucks, but I thought it wasn't a very big deal.

I can rebase and remove these though.

jamesob commented on README.md:
```diff
@@ -76,23 +76,24 @@ $ cd simple-ctv-vault
-#  https://github.com/JeremyRubin/bitcoin/tree/checktemplateverify-rebase-4-15-21
-$ bitcoind -regtest -txindex=1 &
+#  https://github.com/JeremyRubin/bitcoin/tree/checktemplateverify-signet-23.0-alpha
+$ bitcoind -signet -signetchallenge=512102946e8ba8eca597194e7ed90377d9bbebc5d17a9609ab3e35e706612ee882759351ae -signetseednode=50.18.75.225 -txindex=1 &
```
General feedback on this PR: we need to both support signet and regtest, so it would be useful to segment the various parts of this change out and keep regtest-specific content intact.

jamesob commented on main.py:
```diff
@@ -536,16 +533,6 @@ def _print_signed_tx(
-            .get_private_key(1)
-            .point.p2wpkh_address(network=rpc.net_name)
-        )
-    return rpc.generatetoaddress(n, addr)
```
In line with preserving regtest, let's keep this.

jamesob commented on main.py:
```diff
@@ -619,7 +606,6 @@ class VaultScenario:
 
     @classmethod
     def from_network(cls, network: str, seed: bytes, coin: Coin = None, **plan_kwargs):
-        SelectParams(network)
```
Note to self to verify that this doesn't actually do anything

jamesob commented on main.py:
```diff
@@ -648,20 +634,19 @@ def from_network(cls, network: str, seed: bytes, coin: Coin = None, **plan_kwarg
-        c = VaultScenario.from_network(
-            "regtest", seed=b"demo", coin=coin_in, block_delay=10
-        )
+        c = VaultScenario.from_network("signet", b"demo", coin=coin_in, block_delay=10)
```
Hardcoded network constants should be factored out into some kind of `SelectParams`-ish global behavior that still allows use of regtest.

fiatjaf replied:
`SelectParams` wasn't being used for anything as far as I know, and you already had a global registry of ports for each network.

jamesob commented on main.py:
```diff
@@ -677,75 +662,69 @@ def vault():
 def _broadcast_final(c: VaultScenario, tx: CTransaction, hot_or_cold: str):
     print()
-    if hot_or_cold == 'cold':
+    if hot_or_cold == "cold":
```
Would be good to separate out formatting changes if possible.

jamesob commented on main.py:
```diff
@@ -76,6 +77,10 @@
 
 
+def no_output(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
```
We need to revisit this since it definitely isn't well named as "no_output."

fiatjaf replied:
That was the name it had, I was very confused by it :stuck_out_tongue:

jamesob reviewed (commented):
Thanks for working on this. Before something like this is merged, we need to figure out how to preserve optional regtest use. I'll propose a patch set that does this (with you as co-author) unless you get around to it first.

fiatjaf committed:
changes required for running on CTV signet.

- remove useless SelectParams call since bitcoinlib doesn't know signet.
- remove generateblocks, replace it with a message telling the user to get signet coins.
- hardcode demo scenario to signet.
- output the "txid:output" as the original_coin instead of just the txid.

fiatjaf committed:
update readme to default to signet.

fiatjaf committed:
self.log prints to stderr.

fiatjaf commented:
I've removed the whitespace and quote changes and reduced the README changes to a minimum and brought multi-network support back, although the demo still defaults to `signet` and there is no way to change it. I think there could be a global flag for specifying the network but I'm not sure how to do it, so I'll leave that to you.
